### DISCUSSION TITLE
installation / test error

### DISCUSSION THREAD

**Discussion by mch3n** [neuropoly/axondeepseg#648] [Help needed] [answered]
(I'm completely new to axondeepseg)

When I tried running a test, the program keeps giving me: 

File "/Users/margaretchen/axondeepseg/AxonDeepSeg/apply_model.py", line 8, in <module>
    from ivadomed import inference as imed_inference
ModuleNotFoundError: No module named 'ivadomed'

It also keeps giving me an error saying that the axondeepseg file isn't in python format, and that setup.py and pyproject.toml aren't found. 

Below is what it looks like in Terminal.

<img width="1205" alt="Screen Shot 2022-06-10 at 12 32 58 AM" src="https://user-images.githubusercontent.com/107157075/172991286-491a4414-fd07-4693-bbd6-af5b89ae264a.png">

**Comment by hermancollin** [comment, marked as answer]
Hi @mch3n, thank you for reaching out.

Could you please provide the output of `pip freeze` when you are inside the virtual environment?

From what I can see, you were not able to successfully install axondeepseg. The only weird thing I can see is how the files are nested: I see pip is trying to find setup.py in `axondeepseg/AxonDeepSeg/axondeepseg/AxonDeepSeg`, which is weird because the usual file structure is only `axondeepseg/AxonDeepSeg` and this is where setup.py is located. I'm not sure how this happened but I would recommand deleting all these files and start over again.

**Reply by mch3n** [reply]
I'm not sure what happened, but I have a copy of axondeepseg in another axondeepseg folder. Whenever I tried to delete the files and restart the installation process, an error kept occurring, but I played around and somehow got it to successfully download and run. Will the way the files are nested affect the way the program runs now? (So, do I have to fix the way the files are nested, or can I just leave it how it is?)

**Reply by hermancollin** [reply]
@mch3n I don't think the location where you put the axondeepseg folder matters, unless there's a bug we aren't aware of. For this reason, maybe it would be preferable to install the software in a folder with a different name. 

With that being said, if you successfully downloaded the program with `git clone https://github.com/neuropoly/axondeepseg.git` and moved inside the axondeepseg directory with `cd axondeepseg`, what you should see is these files:

![Screenshot_20220612_220437](https://user-images.githubusercontent.com/83031821/173266668-88665c08-accb-438d-b74e-7af4b46ce7f1.png)

You should be able to confirm this using `ls -la` and then proceed with the rest of the instructions to create and activate the virtual environment. Once ADS is correctly installed and the `axondeepseg_test` command runs successfully, you can use the software from any folder that you want (just make sure the virtual environment is activated with `conda activate ads_venv`).
